porting/npl/freertos/include/npl/npl.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _NPL_H_
+#define _NPL_H_
```

Comment:
I preferred when os_<module> function call API was in common headers, and platform defines the specific struct os_<module>.  Is it possible to return to that model?  This scheme has no enforcement of common osal API.

npl.h can define the structs and then include all the individual os_<module>.h headers: os_eventq.h, os_mutex.h, ...

Reply:
I also preferred it that way and will try to fix this. The problem was mostly due to the way NimBLE includes various OS headers and in some cases two files needed different ordering of the same includes to build properly. I made some cleanup in the meantime so it may be better now, will give it a try.

Another small advantage of having declarations in OS NPL is that some functions can be implemented as `static inline`.